tweag/nixpkgs-graph
# Fix for inconsistent json file structure

Fixes #9 

## What this PR does

This fixes the problem of inconsistent structure in the first version of the `edges.json` file causing `pandas` to fail to read it.

### Compared with the version \#6
- Added `type="node"` attribute to the set of each node
- Added `lib.collect` to reclaim all node sets before the final output
- Output all package that cannot be evaluated as `null`

## How to test this PR

- Install nix package package manager if this is not done:
  ```sh
  $ curl -L https://nixos.org/nix/install | sh
  ``` 

- Excute the following command to generate the node information convert its format.
  ```sh
  $ /bin/bash build.sh
  ``` 

- Observe the presence of `edges.json` file in the `rawdata/` folder and a consistent set structure.
